src/test/scala/ie/zalando/fabric/gateway/service/IngressDerivationChainSpec.scala

```diff
@@ -780,6 +802,18 @@ class IngressDerivationChainSpec extends FlatSpec with MockitoSugar with Matcher
     }
   }
 
+  "Annotation Handling" should "allow a subset of annotations through to the generated ingress" in {
+    val ingresses = testableRoutesDerivation(
+      sampleGateway,
+      GatewayMeta(DnsString.fromString("test-gateway").get, "default", None, Map("a" -> "1", "b" -> "2", "c" -> "3")))
+
+    ingresses.foreach { ingress =>
+      ingress.metadata.routeDefinition.additionalAnnotations.keySet should contain("a")
+      ingress.metadata.routeDefinition.additionalAnnotations.keySet should not contain("b")
+      ingress.metadata.routeDefinition.additionalAnnotations.keySet should not contain("c")
```

Comment:
It's not immediately clear why a is allowed and b and c are not. I had to go digging through config to see the hard-coded test config. Is it possible to pass the allowlist into `testableRoutesDerivation` instead?
If not, change the test name to indicate that a is allowed:
``` 
"Annotation Handling" should "only allow the globally configured annotation with key `a`"
```
or rename the key from `a` to `ALLOWED`

Reply:
Unfortunately the config is statically initialized on startup, so it's difficult to change in tests outside of the app properties file. I've updated the test name and variable names.